lib/content-services/src/lib/security/services/security-controls-groups-marks-security.service.spec.ts

```diff
@@ -0,0 +1,306 @@
+        );
+        const response = await service.deleteSecurityMark(securityGroupId, securityMarkId);
+        securityMarkId = response.entry.id;
+        securityGroupId = response.entry.groupId;
+        expect(response.entry.groupId).toEqual('eddf6269-ceba-42c6-b979-9ac445d29a94');
+        expect(response.entry.name).toEqual('securityMark1');
+        expect(response.entry.id).toEqual('ffBOeOJJ');
+    });
+
+    it('should delete a security group', async () => {
+        spyOn(service.groupsApi, 'deleteSecurityGroup').and.returnValue(
+            Promise.resolve(
+                new SecurityGroupEntry({
+                    entry: {
+                        groupName: 'TestGroup',
+                        groupType: 'HIERARCHICAL',
+                        id: 'eddf6269-ceba-42c6-b979-9ac445d29a94'
+                    }
+                })
+            )
+        );
+        const response = await service
+            .deleteSecurityGroup(securityGroupId)
+            .toPromise();
+
+        expect(response.entry.groupName).toEqual('TestGroup');
+        expect(response.entry.groupType).toEqual('HIERARCHICAL');
+        expect(response.entry.id).toEqual('eddf6269-ceba-42c6-b979-9ac445d29a94');
+    });
+
+    it('should be able to get clearances for authority', async () => {
+        const getClearancesForAuthoritySpy = spyOn(
+            service.authorityClearanceApi,
+            'getAuthorityClearanceForAuthority'
+        ).and.returnValue(Promise.resolve(fakeAuthorityClearanceApiResponse));
+        const clearancePromise = service.getClearancesForAuthority('test-id', 0, 10);
+        const clearance = await clearancePromise.toPromise();
+
+        expect(getClearancesForAuthoritySpy).toHaveBeenCalledWith('test-id', {
+            skipCount: 0,
+            maxItems: 10
+        });
+
+        expect(clearance.list.pagination.skipCount).toBe(0);
+        expect(clearance.list.pagination.maxItems).toBe(10);
+        expect(clearance.list.entries[0].entry.id).toBe('test-id');
+        expect(clearance.list.entries[0].entry.displayLabel).toBe('test-displayLabel');
+        expect(clearance.list.entries[0].entry.systemGroup).toBe(false);
+        expect(clearance.list.entries[0].entry.type).toBe('test-type');
+    });
+
+    it('should update a clearances for authority', async () => {
+        spyOn(service.authorityClearanceApi, 'updateAuthorityClearance').and.returnValue(
+            Promise.resolve(
+                new SecurityMarkEntry({
+                    entry: {
+                        id: 'test-id',
+                        name: 'test-name',
+                        groupId: 'test-groupId'
+                    }
+                })
+            )
+        );
+        const response = await service.updateClearancesForAuthority('test-id', [{
+            groupId: 'test-group-id',
+            op: 'test-op',
+            id: 'test-id'
+        }]).toPromise();
+
+        if (response instanceof SecurityMarkEntry) {
```

Comment:
What if implementation of updateClearancesForAuthority will change and it will return SecurityMarkPaging instead of SecurityMarkEntry? Then it won't enter into body of 'if' and none expect will be executed but it should fail in that case. 

To resolve that problem you can for example call done() in body after expects.

Reply:
we discussed on slack that we can't use done here as it is async function